### DISCUSSION TITLE
Add CLI verison to Priority list

### DISCUSSION THREAD

**Discussion by lamyergeier** [TagStudioDev/TagStudio#265] [Ideas]
Please add CLI to [Priority list](https://github.com/TagStudioDev/TagStudio?tab=readme-ov-file#priority-features)

## Observation

I tried to create a new library at a location with about 100 video files. It took a while to load all files and the software was so slow and it kept show "Force close as not responding". May be we really need CLI version of the app.

After the library loaded completely with all thumbnails of the video. I closed it and again opened the library, it again took similar amount of time to load the library!

**Comment by CyanVoxel** [comment]
This has nothing to do with needing a CLI version - this is due to thumbnails not being cached and the program needing to load the video files to grab a frame from on each refresh. Diverting resources towards creating and supporting an entirely new frontend would not be the way to solve this. Implementing a thumbnail caching system (which already is a priority) will significantly improve this process on successive runs, especially for large video files. And even then we could always just add an option to disable thumbnails to the current GUI instead of building a new frontend from the ground up.

**Reply by lamyergeier** [reply]
@CyanVoxel 

Thank you. I accept your points,that CLI is not necessary for the observation that I had raised.

Many people who use terminal and barely use GUI, would be happy with a CLI as CLI is usually faster, uses less resources and can be automated with custom bash scripts. CLI is also quite powerful as users can pipe data and do bulk modifications. 

Thus, I request you to add CLI to priority if possible.
